src/spec-configuration/configuration.ts

```diff
@@ -21,10 +21,17 @@ export type UserEnvProbe = 'none' | 'loginInteractiveShell' | 'interactiveShell'
 
 export type DevContainerConfigCommand = 'initializeCommand' | 'onCreateCommand' | 'updateContentCommand' | 'postCreateCommand' | 'postStartCommand' | 'postAttachCommand';
 
+export interface HostGPURequirements {
+	cores?: number;
+	memory?: string;
+}
+
+
 export interface HostRequirements {
 	cpus?: number;
 	memory?: string;
 	storage?: string;
+	gpu?: boolean | HostGPURequirements;
```

Comment:
It would make more sense to make the "requirement" fail if there is no GPU. E.g., if the `"gpu"` property is set, you would check if a GPU is available and fail if it is not. Only when `"gpu"` is set and a GPU is available would you enable the GPU for the dev container.

This will make sure that dev containers requiring a GPU can count on one being available. (To make the GPU optional but still automatically add it, maybe `"gpu": "optional"` could be added.)

Reply:
@jkeech @Chuxel I realize, we don't enforce any of the other host requirements in the CLI. The thinking was that we don't want to prevent anyone from using the dev container. In the Dev Containers extension we show a warning dialog in the UI if the host requirements are not matched and the user can choose to still continue.

Reply:
@chrmarti Thanks - my motivation for this is all about making it optional, so I'll try and include that.